test/unit/spec/MfaVerify_spec.js

```diff
@@ -2150,13 +2151,28 @@ function (Okta,
                 });
               });
             });
-            itp('on SUCCESS, polling stops and form is submitted', function () {
+            itp('on SUCCESS, sets transaction, polling stops and form is submitted', function () {
               return setupOktaPush().then(function (test) {
+                spyOn(test.router.controller.model, 'setTransaction').and.callThrough();
                 spyOn(test.router.settings, 'callGlobalSuccess');
                 return setupPolling(test, resSuccess)
-                .then(function () { return tick(test); }) // Final tick - SUCCESS
+                .then(function () {
+                  return tick(test);
+                }) // Final tick - SUCCESS
                 .then(function () {
                   expect(test.router.settings.callGlobalSuccess).toHaveBeenCalled();
+                  // One after first poll returns and one after polling finished.
+                  expect(test.router.controller.model.setTransaction.calls.count()).toBe(2);
```

Comment:
we should check that ```setTransaction``` was called with the right arguments. First should be for the MFA_CHALLENGE status, and second one should be for SUCCESS if I'm reading this test correctly